DOC-2062 Global config article for K8s AA public preview (#2490)
RedisLabs/redislabs-docs: content/kubernetes/active-active/preview/global-config.md

---
Title: Set global database configurations
linkTitle: Global configuration
description: The REAADB contains the field '.spec.globalConfigurations' and through this the database configurations are set.
weight: 50
alwaysopen: false
categories: ["Platforms"]
aliases: {
    /kubernetes/active-active/preview/global-config/,

}
---

{{<banner-article bannerColor="#fff8dc">}}
This feature is currently in public preview. Contact Redis support if you plan to use this feature.
See [Create Active-Active databases for Kubernetes]({{<relref "/kubernetes/active-active/create-aa-database.md">}}) for the currently supported procedure.
{{</banner-article>}}

> The Redis Enterprise Active-Active database (REAADB) custom resource contains the field '.spec.globalConfigurations'. This field sets configurations for the Active-Active database across all participating clusters, such as memory size or shard count.

Reviewer: ```suggestion
The Redis Enterprise Active-Active database (REAADB) custom resource contains the field `.spec.globalConfigurations`. This field sets configurations for the Active-Active database across all participating clusters, such as memory size or shard count.
```

Not sure if this is supposed to be back-ticked or not.

The [REAADB API reference](https://github.com/RedisLabs/redis-enterprise-operator/blob/master/deploy/redis_enterprise_active_active_database_api.md) contains a full list of available fields.

## Edit global configurations

1. Edit or patch the REAADB custom resource with your global configuration changes.

  The example command below patches the REAADB named `example-aadb-1` to set the global memory size to 200MB:

  ```sh

>   kubectl patch reaadb example-aadb-1 --type merge --patch /

Author: ```suggestion
  kubectl patch reaadb example-aadb-1 --type merge --patch \
```

  '{"spec": {"globalConfigurations": {"memorySize": "200mb"}}}'
  ```

1. Verify the status is `active` and the spec status is `Valid`.

  ```sh

>   kubectl get reaadb example-aadb-1

Reviewer 2: also here add that we are getting the REDB named `example-aadb-1` or in the doc state that we are going to work on the REDB named `example-aadb-1` for example

  NAME             STATUS   SPEC STATUS   GLOBAL CONFIGURATIONS REDB   LINKED REDBS
  example-aadb-1   active   Valid    
  ```

1. View the global configurations on each participating cluster to verify they are synced.

  ```sh
  kubectl get reaadb <REAADB name> -o yaml
  ```

## Set global database secret

One of the fields available for `globalConfigurations` is `databaseSecretName` which can point to a secret containing the database password.

> ### On an existing participating cluster, do the following

Reviewer: I think instead of having headings - you might want to change these to sentences? The headings seem a little long

1. Generate a YAML file containing the database secret with the database password.

  This example shoes a secret named `my-db-secret` with the password `my-password` encoded in base 64.

  ```yaml
  apiVersion: v1
  data:
    password: bXktcGFzcw
  kind: Secret
  metadata:
    name: my-db-secret
  type: Opaque
  ```

1. Apply the secret file from the previous step, substituting your own value for `<db-secret-file>`.

  ```sh
  kubectl apply -f <db-secret-file>
  ```

1. Patch the REAADB custom resource to specify the database secret.

    ```sh

>     kubectl patch reaadb example-aadb-1 --type merge --patch /

Reviewer 2: also here we should state that the REDB example is named `example-aadb-1`.

    ```sh

>     kubectl patch reaadb <reaadb-name> --type merge --patch /

Author: ```suggestion
    kubectl patch reaadb <reaadb-name> --type merge --patch \
```

1. Check the REAADB status for an `active` status and `Valid` spec status.

    ```sh
    kubectl get reaadb <reaadb-name>

    NAME             STATUS   SPEC STATUS   GLOBAL CONFIGURATIONS REDB   LINKED REDBS
    example-aadb-1   active   Valid
    ```

### On each of the other existing participating clusters, do the following

1. Check secret status.

  ``sh
  kubectl get reaadb <reaadb-name> -o=jsonpath='{.status.secretsStatus}'

  ```

  The output should show the status as `Invalid`.

  ```sh
  [{"name":"my-db-secret","status":"Invalid"}]
  ```

> 1. Sync the secret.

Reviewer 2: I think the numbering is broken in this doc.
